ADAL/src/ADAuthenticationContext+Internal.m

```diff
@@ -101,30 +102,27 @@ + (ADAuthenticationError*)errorFromDictionary:(NSDictionary*)dictionary
                                     errorCode:(ADErrorCode)errorCode
 {
     //First check for explicit OAuth2 protocol error:
-    NSString *serverOAuth2Error = [dictionary objectForKey:OAUTH2_ERROR];
-    if (![NSString adIsStringNilOrBlank:serverOAuth2Error])
+    NSString *serverOAuth2Error = [dictionary adStringForKey:OAUTH2_ERROR];
+    if (serverOAuth2Error)
     {
-        NSString *errorDetails = [dictionary objectForKey:OAUTH2_ERROR_DESCRIPTION];
-        // Error response from the server
-        NSUUID *correlationId = [dictionary objectForKey:OAUTH2_CORRELATION_ID_RESPONSE] ?
-                                [[NSUUID alloc] initWithUUIDString:[dictionary objectForKey:OAUTH2_CORRELATION_ID_RESPONSE]]:
-                                nil;
+        NSString *responseCorrelationId = [dictionary adStringForKey:OAUTH2_CORRELATION_ID_RESPONSE];
+        NSUUID *correlationId = responseCorrelationId ? [[NSUUID alloc] initWithUUIDString:responseCorrelationId] : nil;
 
         ADErrorCode code = errorCode;
-        NSString *suberror = [dictionary objectForKey:AUTH_SUBERROR];
-        NSMutableDictionary *userInfo = nil;
+        NSString *suberror = [dictionary adStringForKey:AUTH_SUBERROR];
+        NSMutableDictionary *userInfo = [NSMutableDictionary new];
+        userInfo[ADSuberrorKey] = suberror;
+
         if (suberror && [suberror isEqualToString:AUTH_PROTECTION_POLICY_REQUIRED])
         {
             code = AD_ERROR_SERVER_PROTECTION_POLICY_REQUIRED;
-            userInfo = [[NSMutableDictionary alloc] initWithCapacity:2];
-            [userInfo setObject:suberror forKey:ADSuberrorKey];
-
-            // check for additional user identifier
-            userInfo[ADUserIdKey] = [dictionary objectForKey:AUTH_ADDITIONAL_USER_IDENTIFIER];
         }
-        
+
+        userInfo[ADUserIdKey] = [dictionary adStringForKey:AUTH_ADDITIONAL_USER_IDENTIFIER];
```

Comment:
kind of goes back to the above question, do we want to introduce another util, 
something like msidSetNonEmptyString or msidSetObjectIfNotNil in common core?

Reply:
Setting nil here doesn't matter, since userInfo is an empty dictionary created above. So, additional check is not necessary here.